Mobile Buy SDK/Mobile Buy SDK Tests/BUYClientTest.m

```diff
@@ -126,9 +126,7 @@ - (void)testPartialAddressesFlag
 
 - (void)testCheckoutPaymentWithOnlyGiftCard
 {
-	BUYCheckout *checkout = [[BUYCheckout alloc] initWithModelManager:self.client.modelManager JSONDictionary:@{@"token": @"abcdef", @"payment_due": @0}];
-	
-	BUYOperation *task = [self.client completeCheckout:checkout paymentToken:nil completion:^(BUYCheckout *checkout, NSError *error) {}];
+	BUYOperation *task = [self.client completeCheckoutForToken:@"abcdef" paymentToken:nil completion:^(BUYCheckout *checkout, NSError *error) {}];
```

Comment:
Many tests in this test case either no longer make sense or simply not needed as a result of having explicit tests for routes in `BUYClient+RoutingTests.m`. I think we'll have to address the cleanup as part of another PR. Thoughts?